core/src/main/java/org/jdbi/v3/core/statement/Call.java

```diff
@@ -164,7 +229,14 @@ public void apply(int outPosition, PreparedStatement statement, StatementContext
             this.position = outPosition;
         }
 
-        public Object map(CallableStatement stmt) {
+        public Supplier<Object> supplier(CallableStatement stmt) {
+            return () -> {
+                Object value = map(stmt);
+                return isNull(stmt) ? null : value;
```

Comment:
if `isNull(stmt)`, does that imply that the `value` we retrieved would always be `null` anyway, which would allow eliding this ternary?

Reply:
IIRC, there are some drivers that e.g. return "0" for an integer column but isNull == true. This is code that already existed and was just moved around. isNull only implies that the SQL column contains a SQL NULL, not that the return value is null.

Reply:
correction, this is true for every driver if e.g. a null is mapped onto an integer so `rs.getInt()` is called. `ResultSet#getInt()` explicitly returns "0" for a SQL NULL (see javadocs) and the `isNull()` call is required to turn it into a `null` value.